File: api/attendance.py
from datetime import datetime
import pandas as pd
import os
import json

def handler(request):
    if request.method != "POST":
        return {
            "statusCode": 405,
            "headers": {"Content-Type": "application/json"},
            "body": json.dumps({"success": False, "message": "Method not allowed"})
        }

    try:
        body = request.json()
        student_number = body.get("student_number")
    except Exception:
        return {
            "statusCode": 400,
            "headers": {"Content-Type": "application/json"},
            "body": json.dumps({"success": False, "message": "Invalid JSON"})
        }

    students_file = "students.json"
    attendance_file = "attendance_records.csv"

    if not os.path.exists(students_file):
        return {
            "statusCode": 404,
            "headers": {"Content-Type": "application/json"},
            "body": json.dumps({"success": False, "message": "students.json not found"})
        }

    with open(students_file, "r") as f:
        students = json.load(f)

    if student_number not in students:
        return {
            "statusCode": 404,
            "headers": {"Content-Type": "application/json"},
            "body": json.dumps({"success": False, "message": "Student not found"})
        }

    student = students[student_number]
    current_date = datetime.now().strftime("%Y-%m-%d")
    current_time = datetime.now().strftime("%H:%M:%S")

    if os.path.exists(attendance_file):
        df = pd.read_csv(attendance_file)
    else:
        df = pd.DataFrame(columns=['Date', 'Time', 'Student Number', 'Student Name', 'Grade Level', 'Section', 'Status'])

    # Prevent duplicate attendance for the same day
    already_marked = df[
        (df['Date'] == current_date) & 
        (df['Student Number'] == student_number)
    ]

    if not already_marked.empty:
        return {
            "statusCode": 200,
            "headers": {"Content-Type": "application/json"},
            "body": json.dumps({"success": True, "message": f"✅ Already marked: {student['name']} ({student_number})"})
        }

    new_record = {
        'Date': current_date,
        'Time': current_time,
        'Student Number': student_number,
        'Student Name': student['name'],
        'Grade Level': student['grade_level'],
        'Section': student['section'],
        'Status': 'Present'
    }

    df = pd.concat([df, pd.DataFrame([new_record])], ignore_index=True)
    df.to_csv(attendance_file, index=False)

    return {
        "statusCode": 200,
        "headers": {"Content-Type": "application/json"},
        "body": json.dumps({"success": True, "message": f"🟢 Marked Present: {student['name']} ({student_number})"})
    }

File: api/attendance.js
import { createClient } from "@supabase/supabase-js";

const supabase = createClient(
  process.env.SUPABASE_URL,
  process.env.SUPABASE_KEY
);

export default async function handler(req, res) {
  if (req.method !== "POST") {
    return res.status(405).json({ success: false, message: "Method not allowed" });
  }

  try {
    const { student_number } = req.body;
    if (!student_number)
      return res.status(400).json({ success: false, message: "Missing student_number" });

    // Load local students.json
    const students = await import("../../students.json", { assert: { type: "json" } });
    const student = students.default[student_number];
    if (!student)
      return res.status(404).json({ success: false, message: "Student not found" });

    const now = new Date();
    const currentDate = now.toISOString().split("T")[0];
    const currentTime = now.toTimeString().split(" ")[0];

    // Check if already marked today
    const { data: existing, error: fetchError } = await supabase
      .from("attendance_records")
      .select("*")
      .eq("date", currentDate)
      .eq("student_number", student_number);

    if (fetchError) throw fetchError;
    if (existing.length > 0) {
      return res.status(200).json({
        success: true,
        message: `✅ Already marked: ${student.name} (${student_number})`,
      });
    }

    // Insert new record
    const { error: insertError } = await supabase.from("attendance_records").insert([
      {
        date: currentDate,
        time: currentTime,
        student_number,
        student_name: student.name,
        grade_level: student.grade_level,
        section: student.section,
        status: "Present",
      },
    ]);

    if (insertError) throw insertError;

    return res.status(200).json({
      success: true,
      message: `🟢 Marked Present: ${student.name} (${student_number})`,
    });
  } catch (err) {
    console.error("Error:", err);
    return res.status(500).json({
      success: false,
      message: "Server error — unable to save attendance",
    });
  }
}
